web-client/src/client/peer_info.rs

```diff
@@ -6,13 +6,49 @@ use wasm_bindgen::prelude::*;
 #[derive(serde::Serialize, Tsify)]
 #[serde(rename_all = "camelCase")]
 pub struct PlainPeerInfo {
+    /// A libp2p peer ID
     pub peer_id: String,
     /// Address of the peer in `Multiaddr` format
     address: String,
     /// Node type of the peer
     #[tsify(type = "'full' | 'history' | 'light'")]
     #[serde(rename = "type")]
     node_type: String,
+    /// List of services the peer is providing
+    services: Vec<PlainService>,
+}
+
+/// Available peer service flags
+#[derive(serde::Serialize, Tsify)]
+#[serde(rename_all = "kebab-case")]
+pub enum PlainService {
+    FullBlocks,
+    History,
+    AccountsProof,
+    AccountsChunk,
+    Mempool,
+    TransactionIndex,
+    Validator,
+    PreGenesisTransactions,
+
+    // Catch-all to not have to panic when new services are added
+    Unknown,
```

Comment:
Is `Unknown` useful info for the web client? If not, I'd prefer to simply skip these.

The `from` function would return an `Option<PlainService>` and the `PlainPeerInfo` could then just ignore the `None`s.

Reply:
I added it because I wanted to avoid silent failing. If we ever add more services, but forget to add them here, noone would ever find out.

Panicking of erroring is bad for backward-compatibility, so I think 'Unknown' is the next best thing.

That's also how I found https://github.com/nimiq/core-rs-albatross/issues/2986, because the new topics were not hidden.